### PR TITLE
Feature/cds 1390 firehose

### PR DESCRIPTION
**Pull request by ryantanjunming**
# Description

 Fixes #

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the relevant example in the examples directory for the module.
- [ ] I have updated the relevant test file for the module.
- [ ] This change does not affect module (e.g. it's readme file change)

### CODE REVIEW

**Review by guyrenny — APPROVED** [review]
LGTM

File: modules/firehose-metrics/output.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+output "metric_stream_arn" {
+  value       = one(aws_cloudwatch_metric_stream.cloudwatch_metric_stream[*].arn)
+  description = "value of the cloudwatch metric stream ARN"
+}
+
+output "metrics_stream_iam_role_arn" {
+  value       = local.metrics_stream_iam_role_arn
+  description = "value of the cloudwatch metric stream IAM role ARN used"
+
+}
```
Comment (guyrenny) [inline modules/firehose-metrics/output.tf:35]: Add empty line in the end

**Review by guyrenny — COMMENTED** [review]
Need to also update the variables.tf file in the examples folder.

File: modules/firehose-metrics/README.md [hunk 2]
```diff
@@ -192,33 +192,39 @@ then the CloudWatch metric stream must be configured with the same format, confi
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
 | <a name="input_coralogix_region"></a> [coralogix\_region](#input\_coralogix\_region) | Coralogix account region: Europe, Europe2, India, Singapore, US, US2 [exact] | `any` | n/a | yes |
-| <a name="input_private_key"></a> [private_key](#input\_private_key) | Coralogix account logs private key | `any` | n/a | yes |
+| <a name="input_private_key"></a> [private\_key](#input\_private_key) | Coralogix account logs private key | `any` | n/a | yes |
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply modules/firehose-metrics/README.md:195]: okay changing

File: modules/firehose-metrics/main.tf [hunk 3]
```diff
@@ -25,10 +25,22 @@ locals {
...
-  lambda_processor_name          = var.lambda_processor_custom_name != null ? var.lambda_processor_custom_name : "${var.firehose_stream}-metrics-transform"
+  # global resource referecing
+  s3_backup_bucket_arn          = var.existing_s3_backup != null ? one(data.aws_s3_bucket.exisiting_s3_bucket[*].arn) : one(aws_s3_bucket.new_s3_bucket[*].arn)
+  lambda_processor_iam_role_arn = var.existing_lambda_processor_iam != null ? one(data.aws_iam_role.existing_lambda_iam[*].arn) : one(aws_iam_role.new_lambda_iam[*].arn)
+  firehose_iam_role_arn         = var.existing_firehose_iam != null ? one(data.aws_iam_role.existing_firehose_iam[*].arn) : one(aws_iam_role.new_firehose_iam[*].arn)
+  metrics_stream_iam_role_arn   = var.existing_metric_streams_iam != null ? one(data.aws_iam_role.existing_metric_streams_iam[*].arn) : one(aws_iam_role.new_metric_streams_iam[*].arn)
+
+  # default resource namings
+  lambda_processor_name          = var.lambda_processor_custom_name != null ? var.lambda_processor_custom_name : "${var.firehose_stream}-metrics-transform-${random_string.this.result}"
+  firehose_stream_name           = "${var.firehose_stream}-metrics-${random_string.this.result}"
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply modules/firehose-metrics/main.tf]: removed and made it firehose_stream

File: modules/firehose-metrics/output.tf (hunk 1 above)
Comment (ryantanjunming) [reply modules/firehose-metrics/output.tf:34]: removed

File: modules/firehose-metrics/README.md [hunk 4]
```diff
@@ -192,33 +192,39 @@ then the CloudWatch metric stream must be configured with the same format, confi
...
-| <a name="input_user_supplied_tags"></a> [user_supplied_tags](#input\_user_supplied_tags) | Tags supplied by the user to populate to all generated resources | `map(string)` | n/a | no |
-| <a name="input_override_default_tags"></a> [override_default_tags](#input\_override_default_tags) | Override and remove the default tags by setting to true | `bool` | `false` | no |
+| <a name="input_additional_metric_statistics_enable"></a> [additional\_metric\_statistics\_enable](#input\_additional\_metric\_statistics\_enable) | To enable the inclusion of additional statistics to the streaming metrics | `bool` | `true` | no |
+| <a name="input_additional_metric_statistics"></a> [additional\_metric\_statistics](#input\_additional\_metric\_statistics) | For each entry, specify one or more metrics (metric_name and namespace) and the list of additional statistics to stream for those metrics. Each configuration of metric name and namespace can have a list of additional_statistics included into the AWS CloudWatch Metric Stream. | `list(object({additional_statistics=list(string), metric_name=string, namespace=string}))` | See variables.tf | no |
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply modules/firehose-metrics/README.md]: okay done

**Comment by gregops312** [comment]
I have to ask why this change was released and you didn't release a major version change, since after all swapping arguments is a breaking change to your consumers. I am specifically referring to moving from `private_key` to `api_key`in [firehose-metrics](https://github.com/coralogix/terraform-coralogix-aws/tree/master/modules/firehose-metrics)

And the documentation for the module has no note of the change, its only visible in the commit, and never made it in the release notes.

**Comment by ryantanjunming** [comment]
Hi @gregops312, many apologies I should have written it in the PR, it was documented in the ChangeLog.md: https://github.com/coralogix/terraform-coralogix-aws/blob/feature/cds-1390-firehose/CHANGELOG.md#v10105

There will also be another change to the `coralogix_region` soon to follow [https://coralogix.com/docs/user-guides/account-management/account-settings/coralogix-domain/](https://coralogix.com/docs/user-guides/account-management/account-settings/coralogix-domain/)

**Comment by gregops312** [comment]
> Hi @gregops312, many apologies I should have written it in the PR, it was documented in the ChangeLog.md: https://github.com/coralogix/terraform-coralogix-aws/blob/feature/cds-1390-firehose/CHANGELOG.md#v10105
> 
> There will also be another change to the `coralogix_region` soon to follow https://coralogix.com/docs/user-guides/account-management/account-settings/coralogix-domain/

Thanks for that, I stupidly didn't actually look at the `ChangeLog.md` that is definitely on me. Thanks I will put a note in our usage so we are ready for when that changes.

**Pull request merged**